### PR TITLE
[UI] Import using URL added in filters

### LINKED ISSUE #3745: Add option to import WASM filters via a URL
Related issues: #3746 #3747 

#### Current Behavior

Users are able to upload WASM filters through the "Filters" page in the UI from their local storage. But Meshery also supports fetching files via a provided URL.

#### Desired Behavior

The UI should have an option for the user to fetch filters by providing a URL. The design could be similar to how Google Docs handle importing images.

Screenshot 2021-08-02 at 8 17 36 PM

#### Implementation

This consideration should be made consistent across filters, patterns and applications constructs.

[Nice to have] A design of how the UI would look like should be made prior to implementing this design.

---
**[Contributor Guide](https://docs.meshery.io/project/contributing)**

### PR DESCRIPTION
**Pull request by meghanacosmos**
**Description**

This PR fixes #3745

**Notes for Reviewers**
Currently just using the function used for file upload for URL upload too. 

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: ui/components/Filters.js [hunk 1]
```diff
@@ -435,7 +447,7 @@ function MesheryFilters({ updateProgress, enqueueSnackbar, closeSnackbar, user,
         <YAMLEditor filter={selectedRowData} onClose={resetSelectedRowData()} onSubmit={handleSubmit} />
       )}
       <MUIDataTable
-        title={<div className={classes.tableHeader}>Filters</div>}
+        title={<div className={classes.tableHeader}>Filters?</div>}
```
Comment (leecalcote) [inline ui/components/Filters.js:450]: I might be missing context, I'm trying to think of why we would want a question mark here, though...￼

Comment (Meghana-12) [reply ui/components/Filters.js:450]: Ahh... sorry about that, I added that at the very beginning while playing around with the code. I'll remove that :D

File: ui/components/URLUploader.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  const [open, setOpen] = React.useState(false);
+  const [input, setInput] = React.useState('');
+  const handleOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+  const validURL = (str) =>  {
```
[the remark this answers is not in the document]
Comment (dhruv0000) [reply ui/components/URLUploader.js:35]: Yes, a good place to define this kind of functions would be https://github.com/meshery/meshery/tree/master/ui/utils @Meghana-12

[the remark this answers is not in the document]
Comment (leecalcote) [reply ui/components/URLUploader.js:35]: Note, that we already have url validation in the performance test form.

[the remark this answers is not in the document]
Comment (leecalcote) [reply ui/components/URLUploader.js:35]: @dhruv0000, @sudo-NithishKarthik, we're in need of centralizing this. Will you take an issue to do so?

File: ui/components/Filters.js (hunk 1 above)
Comment (leecalcote) [reply ui/components/Filters.js:450]: No sweat. Resolve this comment when you have removed it.

**Comment by leecalcote** [comment]
@Meghana-12 there are merge conflicts to address.

File: ui/components/URLUploader.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    position: 'absolute',
+    width: 600,
+    backgroundColor: theme.palette.background.paper,
+    border: '0px solid #000',
+    boxShadow: theme.shadows[5],
+    padding: theme.spacing(2, 4, 3),
+    top: '50%',
+    left: '50%',
+    transform: `translate(-50%, -50%)`,
+    borderRadius: 10,
```
[the remark this answers is not in the document]
Comment (Tanuj22) [reply ui/components/URLUploader.js:16]: Need to look into a good way to create a centralized theme. Currently, there is no centralized theme, right @dhruv0000?

File: ui/components/URLUploader.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sudo-NithishKarthik) [reply ui/components/URLUploader.js:35]: But it is written as a method for that class. So, will you just add it in the utils folder as @dhruv0000  mentioned and use it in both places ?

[the remark this answers is not in the document]
Comment (Meghana-12) [reply ui/components/URLUploader.js:35]: Sure, I'll do that.

**Comment by leecalcote** [comment]
So close to being done...

**Comment by leecalcote** [comment]
Making a release today... we'll catch this PR on the next release.

**Comment by leecalcote** [comment]
When I build this branch, I get a UI that is missing the Configuration menu.

Screen Shot 2021-08-19 at 11 37 08 PM

**Comment by Meghana-12** [comment]
> When I build this branch, I get a UI that is missing the Configuration menu.
> 
> Screen Shot 2021-08-19 at 11 37 08 PM
Oh, that's weird, I didn't change anything related to that. Have you selected the None provider? I think Configurations is only available for None.

**Comment by Meghana-12** [comment]
Should I fix the patterns and application issues in this PR itself or raise another PR?

**Comment by sudo-NithishKarthik** [comment]
> Should I fix the patterns and application issues in this PR itself or raise another PR?

Raising a new PR will be good.

**Comment by sudo-NithishKarthik** [comment]
> When I build this branch, I get a UI that is missing the Configuration menu.

@leecalcote , can you check this once again, it is working fine for me

**Pull request merged**